tools/azsdk-cli/docs/sdk-sample-generation-plan.md

```diff
@@ -0,0 +1,387 @@
+# Plan: Add "SDK Sample Generation" command to azsdk-cli
+
+This document outlines the changes to add a new CLI/MCP tool that generates SDK samples per language. The command will live under the existing Generators command group as `azsdk generators samples` and expose an MCP method for agents.
+
+## Goals and scope
+
+- CLI command to generate sample code for an Azure SDK package based on user prompts.
+- Support per-language conventions (dotnet, java, js/ts, python, go) across Azure SDK mono repositories.
+- Default to auto-detecting the language and repo from `--package-path`.
+- AI-powered generation via existing MicroagentHostService (Azure OpenAI) to create samples from user descriptions.
+- Safe-by-default: dry-run and overwrite controls.
+
+**Supported Azure SDK Repositories:**
+- .NET: [azure-sdk-for-net](https://github.com/Azure/azure-sdk-for-net)
+- Java: [azure-sdk-for-java](https://github.com/Azure/azure-sdk-for-java)
+- JavaScript/TypeScript: [azure-sdk-for-js](https://github.com/Azure/azure-sdk-for-js)
+- Python: [azure-sdk-for-python](https://github.com/Azure/azure-sdk-for-python)
+- Go: [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go)
+
+## High-level design
+
+- New tool class `SampleGeneratorTool : MCPTool`
+  - Command path: `azsdk generators samples`
+  - Options to locate package, select language, provide user prompt for sample generation, output location, dry-run, overwrite, and AI model.
```

Comment:
If not in `dry-run` mode, what permanent effect would the tool have. In other words, should the output always be easily reversible or tweakable even without a `dry-run`?